### PR TITLE
chore+feat: local dev fixes

### PR DESCRIPTION
**Pull request by superlinkx**
## Description
Local Dev Improvements for the following:
-   Implement `cosmtrek/air` for our Go hot reloads
-   Add better local configuration options to the docker-compose.yml files
-   Pathway for Docker Desktop on Mac to run hot reloads correctly
-   Proper dev documentation for this setup
-   Clean up and deduplicate as much as possible
-   Documentation available at https://github.com/SpecterOps/BloodHound/wiki/Local-Development-with-Docker-Compose

## Motivation and Context

Several painpoints remain for many devs on the team, so this is meant to resolve all common pain points and provide the missing documentation to give devs better agency over their environment.

## How Has This Been Tested?
- Local testing by me
- Local testing by several other engineers

## Types of changes

- [x] Chore (a change that does not modify the application functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] Documentation updates are needed, and have been made accordingly.
- [ ] I have added and/or updated tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My changes include a database migration.

### CODE REVIEW

**Review by zinic — APPROVED** [review]
Nothing stands out. Solid changeset.

File: cmd/ui/public/img/logo-white-full.svg
Comment (juggernot325) [inline cmd/ui/public/img/logo-white-full.svg:1]: I'm not sure why this file is included in the changeset. I loaded up the apps locally and everything is loading fine, so it's probably not an issue. Still weird, though.

**Review by juggernot325 — APPROVED** [review]
I pulled this branch down and tested locally on an M1 Mac and all is working well! Thank you for this change it's going to make local dev much less tedious.

File: cmd/ui/public/img/logo-white-full.svg
Comment (elikmiller) [reply cmd/ui/public/img/logo-white-full.svg:1]: The license was added to this file - you can click "Display the source diff" and validate

**Pull request merged**